packages/server/graphql/types/Discussion.ts

```diff
@@ -42,6 +49,70 @@ const Discussion = new GraphQLObjectType<any, GQLContext>({
       description:
         'The partial foregin key that describes the type of object that is the topic of the discussion. E.g. AgendaItem, TaskId, ReflectionGroup, GitHubIssue'
     },
+    stage: {
+      type: NewMeetingStage,
+      description: 'The stage that the discussion is located',
+      resolve: async (
+        {discussionTopicId, discussionTopicType, meetingId},
+        _args: unknown,
+        {dataLoader}
+      ) => {
+        const meeting = await dataLoader.get('newMeetings').load(meetingId)
+        const {phases, teamId} = meeting
+        switch (discussionTopicType) {
+          case 'agendaItem': {
+            const phase = phases.find((phase) => phase.phaseType === AGENDA_ITEMS) as
+              | AgendaItemsPhase
+              | undefined
```

Comment:
+1 if you'd like to get away without casting, you can create type predicates like
```
export const isAgendaItemsPhase = (phase: GenericMeetingPhase): phase is AgendaItemsPhase =>
  phase.phaseType === 'agendaitems'
```

and use it like 
```
const phase = phases.find(isAgendaItemsPhase)
```

and you can do all the phase types that way. we already have some predicates in `meetingTypePredicats.ts` file. you can add more there

Reply:
TIL!